enhancements/etcd/etcd-encyrption-for-separate-oauth-resources.md

```diff
@@ -0,0 +1,102 @@
+---
+title: ETCD-Encryption-With-Separate-OAuth-API-Resources
+authors:
+  - "@p0lyn0mial"
+reviewers:
+  - "@sttts"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2020-03-17
+last-updated: 2020-03-17
+status: implementable
+see-also: https://github.com/openshift/enhancements/blob/master/enhancements/authentication/separate-oauth-resources.md, https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/
+replaces:
+superseded-by:
+---
+
+# ETCD Encryption With Separate OAuth API-Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+
+## Motivation
+
+
+### Goals
+1. Make it possible to run the OpenShift OAuth API Server on an encrypted cluster.
+
+### Non-Goals
+
+
+## Proposal
+
+In OpenShift encryption at rest leverages Kubernetes built-in mechanism. It is based on `EncryptionConfiguration` that controls how API data is encrypted in etcd.
+It holds an array of keys that are used to encrypt resources. 
+
+OpenShift maintains a single `EncryptionConfiguration` resource that is created and maintained by a set of controllers called the encryption controllers in `openshift-config-managed` namespace for each component that needs encryption.
+
```

Comment:
before going into detail, first state what we do high-level: for one release `n` let openshift-apiserver-operator maintain the encryption config for both oauth and openshift-apiserver, in order to support downgrades. Version `n+1` then will copy over the config. From that point on authentication-operator will maintain its own config. The authentication-operator will create necessary key secret automatically (no need to copy those).